internal/pkg/fieldtags/analyzer.go

```diff
@@ -71,73 +71,81 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return
 		}
 		for _, f := range (*s).Fields.List {
-			if patterns.isSource(f) && len(f.Names) > 0 {
-				fNames := make([]string, len(f.Names))
-				for i, ident := range f.Names {
-					fNames[i] = ident.Name
-					taggedFields[pass.TypesInfo.ObjectOf(ident)] = true
+			tags := extractTags(f)
+			isTaggedField := false
+			for _, ft := range tags {
+				if conf.IsSourceFieldTag(ft.key, ft.val) {
+					isTaggedField = true
 				}
-				pass.Reportf(f.Pos(), "tagged field: %s", strings.Join(fNames, ", "))
 			}
+			if !isTaggedField || len(f.Names) == 0 {
+				continue
+			}
+			fNames := make([]string, len(f.Names))
+			for i, ident := range f.Names {
+				fNames[i] = ident.Name
+				taggedFields[pass.TypesInfo.ObjectOf(ident)] = true
+			}
+			pass.Reportf(f.Pos(), "tagged field: %s", strings.Join(fNames, ", "))
 		}
 	})
 
 	return ResultType(taggedFields), nil
 }
 
-func (sp *sourcePatterns) isSource(field *ast.Field) bool {
+func extractTags(field *ast.Field) (tags []fieldTag) {
 	if field.Tag == nil {
-		return false
+		return
 	}
 
-	tag := field.Tag.Value
+	wholeTag := field.Tag.Value
 
 	// TODO: consider refactoring this logic into a regex matcher
 	i := 1 // skip initial quote
 	j := 1
-	for j < len(tag) {
-		for j < len(tag) && tag[j] != ':' {
+	for j < len(wholeTag) {
+		for j < len(wholeTag) && wholeTag[j] != ':' {
 			j++
 		}
-		key := tag[i:j]
+		key := wholeTag[i:j]
 		if key == "" {
-			return false
+			return
 		}
 
 		i = j + 1 // skip colon
-		if i >= len(tag) {
-			return false
+		if i >= len(wholeTag) {
+			return
 		}
-		if tag[i] == '\\' {
+		if wholeTag[i] == '\\' {
 			i++
 		}
 		i++ // skip quote
 
 		j = i
-		for j < len(tag) && tag[j] != '"' {
+		for j < len(wholeTag) && wholeTag[j] != '"' {
 			// skip escape character
-			if tag[j] == '\\' {
+			if wholeTag[j] == '\\' {
 				j++
 			}
 			j++
 		}
-		value := tag[i:j]
+		value := wholeTag[i:j]
+		// remove trailing escaped quote if present
+		value = strings.TrimSuffix(value, `\"`)
```

Comment:
Highlighting this change with a comment since it is lost in a sea of renamings.

Reply:
I think I'm going to play the regexp golf for this method after this gets submitted.

Reply:
Sounds like fun, curious to see what you come up with :smile: